testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -12438,3 +12525,142 @@ fn signer_can_accept_rejected_block() {
+/// Miner 1 successfully mines block N+2 with transfer tx (this is mainly for ensuring everything still works after the expiration time)
+/// Sortition occurs. Miner 2 wins.
+/// Miner 2 successfully mines block N+3 including the contract-call previously rejected by miner 1
+/// Ensures both the miners are aligned
+#[test]
+#[ignore]
+fn miner_rejection_by_contract_call_execution_time_expired() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    let num_signers = 5;
+    let num_txs = 3;
+
+    let mut miners = MultipleMinerTest::new_with_config_modifications(
+        num_signers,
+        num_txs,
+        |signer_config| {
+            // Lets make sure we never time out since we need to stall some things to force our scenario
+            signer_config.block_proposal_validation_timeout = Duration::from_secs(1800);
+            signer_config.tenure_last_block_proposal_timeout = Duration::from_secs(1800);
+            signer_config.first_proposal_burn_block_timing = Duration::from_secs(1800);
+        },
+        |config| config.miner.max_execution_time_secs = Some(0),
+        |config| config.miner.max_execution_time_secs = None,
+    );
+    let rl1_skip_commit_op = miners
+        .signer_test
+        .running_nodes
+        .counters
+        .naka_skip_commit_op
+        .clone();
+    let rl2_skip_commit_op = miners.rl2_counters.naka_skip_commit_op.clone();
+
+    let (conf_1, _) = miners.get_node_configs();
+    let (miner_pkh_1, miner_pkh_2) = miners.get_miner_public_key_hashes();
+    let (_miner_pk_1, miner_pk_2) = miners.get_miner_public_keys();
+
+    info!("------------------------- Pause Miner 2's Block Commits -------------------------");
+
+    // Make sure Miner 2 cannot win a sortition at first.
+    rl2_skip_commit_op.set(true);
+
+    miners.boot_to_epoch_3();
+
+    let burnchain = conf_1.get_burnchain();
+    let sortdb = burnchain.open_sortition_db(true).unwrap();
+
+    info!("------------------------- Pause Miner 1's Block Commits -------------------------");
+    rl1_skip_commit_op.set(true);
+
+    info!("------------------------- Miner 1 Mines a Nakamoto Block N -------------------------");
+    miners
+        .mine_bitcoin_block_and_tenure_change_tx(&sortdb, TenureChangeCause::BlockFound, 60)
+        .expect("Failed to mine BTC block followed by Block N");
+
+    miners.wait_for_test_observer_blocks(60);
+
+    miners.send_fee = 300;
+    // First, lets deploy the contract
+    let dummy_contract_src = "(define-public (dummy (number uint)) (begin (ok (+ number u1))))";
+
+    let contract_publish_txid = miners
+        .send_and_mine_contract_publish("dummy-contract", dummy_contract_src, 60)
+        .expect("Failed to publish contract in a new block");
+
+    miners.wait_for_test_observer_blocks(60);
+
+    assert_eq!(last_block_contains_txid(&contract_publish_txid), true);
```

Comment:
Related to my previous comments, these could all be combined into the `send_and_mine_contract_publish` function.